common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/rest/RestClientVcdBasicAuthInterceptor.java

```diff
@@ -60,9 +60,9 @@ public class RestClientVcdBasicAuthInterceptor extends RestClientRequestIntercep
 	private MediaType contentType;
 
 	/**
-	 * session api path.
+	 * Provider session api path.
 	 */
-	private static final String URL_SESSION = "/api/sessions";
+	private static final String PROVIDER_URL_SESSION = "/cloudapi/1.0.0/sessions/provider";
 	
```

Comment:
I'd suggest to make both the REST endpoint backward compatible with old API version, for example check the vCD API version and if it is >= 38.0 then use the PROVIDER_URL_SESSION  otherwise to use URL_SESSION

Reply:
Thank you for your feedback.
According to the API documentation is available since API v33 - https://developer.vmware.com/apis/vmware-cloud-director/latest/cloudapi/1.0.0/sessions/provider/post/

From these docs we can see v33 is mapped to VCD 10, which is deprecated -> https://developer.vmware.com/docs/14143/vmware-cloud-director-api-programming-guide/

Should we support deprecated versions?

Reply:
I think @akantchev 's idea is to have support for backwards compatibility in case the user is running on an older vCD version which supports just the old api

Reply:
I see the second link isn't working properly - I am attaching another one - https://docs.vmware.com/en/VMware-Cloud-Director/index.html
VCD API version <=34 is deprecated and not supported, and this change supports =>33

Reply:
changing should be fine.  The old endpoint will be for very old VCD versions that are out of support for more than an year (10.1 (v34) EOL is 2022-04-09)